source/nijigenerate/core/settings.d

```diff
@@ -18,24 +19,51 @@ string incSettingsPath() {
     return buildPath(incGetAppConfigPath(), "settings.json");
 }
 
+string incSettingsMoveCorruptedFile(string path) {
+    import std.uuid : randomUUID;
+    // move the corrupted settings file to a new location
+    string backupPath = path ~ "." ~ randomUUID().toString();
```

Comment:
yyyymmddHHMMSS.sss might be easier to see which one is the latest one if failed to serialize multiple times.